src/verifier.ts

```diff
@@ -220,26 +220,86 @@ program.command('generate-trace', 'Generate a transaction trace file using the p
     .option('--chain <id>', '<id> of the chain (0 for pre-EIP-155 semantics, 1 for mainnet)', program.INTEGER, 0)
     .option('--transactions <number>', '<number> of transactions to include in the trace', program.INTEGER, undefined, true)
     .option('--file <path>', '<path> to output file', program.STRING, undefined, true)
+    .option('--contract <contract>', 'A contract from ["omg", "bittrex", "etherdelta"] for the transaction to call', program.STRING, undefined, false)
     .action(async (a, o, l) => {
+        
+
         const nonceMap = new Map<bigint, bigint>();
         const ws = fs.createWriteStream(o['file']);
 
         for (let i = 0; i < o['transactions']; i++) {
             const toAccountNum = BigInt(Math.floor(Math.random() * o['toAccounts']) + 1); // random account between 1-toAccounts
             const fromAccountNum = BigInt(Math.floor(Math.random() * (o['fromAccountsEnd'] - o['fromAccountsStart'] + 1) + o['fromAccountsStart'])); // random account between fromAccountsStart - fromAccountsEnd
             
-            const to = await getPublicAddress(toAccountNum);
+            var to = await getPublicAddress(toAccountNum);
+            var value = BigInt(o['value']);
             
             const nonce = nonceMap.has(fromAccountNum) ? nonceMap.get(fromAccountNum)! + 1n : 0n;
             nonceMap.set(fromAccountNum, nonce); 
+        
+            var txnData = Buffer.from([]);
+            if (o['contract']) {
+                switch(o['contract']) {
+                    case "omg":
+                        var addr1 = to;
+                        value = BigInt(0); // Txn value for this contract must be 0
+
+                        const contractvalue = BigInt(Math.floor(Math.random() * 10000 + 1));
+                        var valueStr = contractvalue.toString(16);
+                        // Pad valueStr with zeroes
+                        while (valueStr.length < 64) {
+                            valueStr = '0' + valueStr;
+                        }
+                        var addr1Str = addr1.toString(16);
+                        // Pad with zeroes
+                        while (addr1Str.length < 64) {
+                            addr1Str = '0' + addr1Str;
+                        }
+                        // This is the Ethereum address of the OMGToken contract
+                        to = BigInt('0xd26114cd6EE289AccF82350c8d8487fedB8A0C07');
```

Comment:
Use a literal: `0xd26114cd6EE289AccF82350c8d8487fedB8A0C0n`

But I guess keep this, per our discussion make it a parameter

Reply:
done